airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/DefaultJobPersistence.java

```diff
@@ -340,6 +340,14 @@ private Optional<Job> getJobOptional(final DSLContext ctx, final long jobId) {
     return getJobFromResult(ctx.fetch(BASE_JOB_SELECT_AND_JOIN + "WHERE jobs.id = ?", jobId));
   }
 
+  @Override
+  public Long getJobCount(final Set<ConfigType> configTypes, final String connectionId) throws IOException {
+    final Result<Record> result = jobDatabase.query(ctx -> ctx.fetch(
+        "SELECT COUNT(*) FROM jobs WHERE CAST(config_type AS VARCHAR) in " + Sqls.toSqlInFragment(configTypes) + " AND scope = '" + connectionId
```

Comment:
Any reason why we are not using jooq to generate the query here?

Reply:
Not any particularly good reason -- I was mainly just following the trend of the other job query implementations. But using jooq is definitely more maintainable, so I can switch to that instead 👍

Reply:
I've refactored the full queries that I added into jooq and they look a lot better now. Thanks for the suggestion!